scripts/fit.py

```diff
@@ -275,30 +275,44 @@ def _fit(N_PARS=3, sign=1):
             )
         return result
 
+    def get_chi2ndf(result, xVal):
+        return result.fun / (len(xVal) - len(result.x))
+
     result = _fit(N_PARS)
-    if not result.success:
-        print(result)
+    chi2ndf = get_chi2ndf(result, xVal)
+    best_result = result
+    if not result.success or chi2ndf > max_chi2ndf:
         print("\n\tRetry fitting with new start point!\n")
         result = _fit(N_PARS, -1)
+        chi2ndf = get_chi2ndf(result, xVal)
+        if chi2ndf < get_chi2ndf(best_result, xVal):
+            best_result = result
 
     # repeat fit with less complex model
-    while not result.success:
-        print(result)
+    while not result.success or chi2ndf > max_chi2ndf:
```

Comment:
`result.success` has probably much tighter requirements than a maximum chi2/ndf than 25 or 10. Mayxbe it might be beneficial to try a couple of more times for the fit and only fall back to the chi2 requirement if nothing else works?

Reply:
Actually not. That was the case for the (failing) fit in the Z+Jet bin.